apps/src/templates/sectionProgressV2/IconKey.jsx

```diff
@@ -8,8 +8,13 @@ import styles from './progress-table-legend.module.scss';
 import {Heading6} from '@cdo/apps/componentLibrary/typography';
 import FontAwesome from '../FontAwesome';
 
-export default function IconKey({isViewingLevelProgress, hasLevelValidation}) {
+export default function IconKey({isViewingValidatedLevel, expandedLessonIds}) {
   const [isOpen, setIsOpen] = useState(false);
+  const [isViewingLevelProgress, setIsViewingLevelProgress] = useState(false);
+
+  React.useEffect(() => {
+    setIsViewingLevelProgress(expandedLessonIds.length > 0);
+  }, [expandedLessonIds]);
```

Comment:
I don't think this case calls for a `useEffect`. There's no side effect here, it's just a very simple bit of logic to determine rendering within this component.

```suggestion
  const isViewingLevelProgress = expandedLessonIds.length > 0;
```